code/go/0chain.net/smartcontract/stakepool/stakepool.go

```diff
@@ -520,19 +520,37 @@ func (sp *StakePool) getRandPools(seed int64, n int) []*DelegatePool {
 		return pls
 	}
 
-	// get random N from pools N
-	plsIdxs := rand.New(rand.NewSource(seed)).Perm(n)
-	selected := make([]*DelegatePool, 0, n)
+	var plsIdxs []int
+	var selected []*DelegatePool
 
-	for _, idx := range plsIdxs {
-		selected = append(selected, pls[idx])
-	}
+	_ = cstate.WithActivation(balances, "demeter", func() error {
+		plsIdxs = rand.New(rand.NewSource(seed)).Perm(n)
+		selected = make([]*DelegatePool, 0, n)
+
+		for _, idx := range plsIdxs {
+			selected = append(selected, pls[idx])
+		}
+		return nil
+	}, func() error {
+		plsIdxs = rand.New(rand.NewSource(seed)).Perm(len(sp.Pools))
```

Comment:
you can do
```
plsIdxs = rand.New(rand.NewSource(seed)).Perm(len(sp.Pools))[:n]
```
 and use the previous version of code, so you can hide only this line under hardfork to redefine the plsIdxs list